### DISCUSSION TITLE
The future of Allow bind mounts of nodev,nosuid,noexec filesystems issue

### DISCUSSION THREAD

**Discussion by hadrabap** [opencontainers/runc#4110] [General]
Hello,

I'm a bit interested into the [Allow bind mounts of nodev,nosuid,noexec filesystems](https://github.com/opencontainers/runc/pull/3805). This issue blocks KinD and K3D in rootless Podman.

I've seen the issue is scheduled for the 1.2.0 milestone. I have a few questions in this regard:

1. When is the 1.2.0 milestone scheduled to happen?
2. Will be the 1.2.0 version backward compatible (drop-in replacement in best case scenario) with the 1.1.x version?
3. Is it possible to back-port the issue into the 1.1.x release so downstream projects can just "upgrade" their dependency in runc?

Thank you very much for your time.

PETR

**Comment by AkihiroSuda** [comment]
> When is the 1.2.0 milestone scheduled to happen?

No schedule, hope we can release it by the end of the year...

- https://github.com/opencontainers/runc/issues/4114

> Will be the 1.2.0 version backward compatible (drop-in replacement in best case scenario) with the 1.1.x version?

Yes. If you find some incompatibility, please notify us.

>Is it possible to back-port the issue into the 1.1.x release so downstream projects can just "upgrade" their dependency in runc?

Probably too big to cherry-pick, but I think this one is negotiable.

**Comment by cyphar** [comment]
It should be noted that the code in the PR you linked was quite significantly reworked in https://github.com/opencontainers/runc/pull/3967 (also slated to be in 1.2.0).

It really depends on what you mean by "backwards compatible". There are certain cases where runc had broken behaviour and those have been fixed, so if you depended on the broken behaviour (i.e. behaviour where many flags were ignored when configuring bind-mounts) then you would have issues. But it seems very unlikely anyone depended on the broken behaviour, because it means you were setting an option you didn't want and didn't care that it was ignored. Sane configurations will work the same as before though.

As for backporting, the changes are a bit too complicated to backport and we do not intend to maintain the 1.1.z branch after the 1.2.0 release, so backporting it when we plan to release 1.2.0 makes little sense IMHO.

**Reply by hadrabap** [reply]
Thanks, I will test new snapshot of `main`.

**Comment by hadrabap** [comment]
OK, thank you very much for your answers. I think the best way for me is to wait for the 1.2.0 release and don't create additional work. :-)
